tests/integration/pilot/revisioned_upgrade_test.go

```diff
@@ -103,16 +125,21 @@ func testUpgradeFromVersion(ctx framework.TestContext, fromVersion string) {
 		}
 	}
 
-	sendSimpleTrafficOrFail(ctx, revisionedInstance)
-}
+	// Stop the traffic generator.
+	close(stop)
 
-// sendSimpleTrafficOrFail sends an echo call to the upgrading echo instance
-func sendSimpleTrafficOrFail(ctx framework.TestContext, i echo.Instance) {
-	ctx.Helper()
-	apps.PodA[0].CallWithRetryOrFail(ctx, echo.CallOptions{
-		Target:   i,
-		PortName: "http",
-	})
+	// Get the result.
+	r, err := g.WaitForResult(waitTimeout)
+	if err != nil {
+		ctx.Fatalf("failed waiting for traffic result: %v", err)
+	}
+
+	if r.PercentSuccess() < successThreshold {
+		ctx.Fatalf("percentage of successful calls %f less than threshold %f:\n%s",
+			r.PercentSuccess(), successThreshold, r)
+	}
+
+	scopes.Framework.Infof("traffic test successful!\n%s", r)
```

Comment:
nit: shouldwe log the success rate? if we see its always >95% maybe we bump up the threshold